File: peacedesign/src/main/java/com/peacedesign/android/utils/touchutils/HoverPushOpacityEffect.java
package com.peacedesign.android.utils.touchutils;

import static com.peacedesign.android.utils.touchutils.HoverPushEffect.Pressure.MEDIUM;

import android.animation.ValueAnimator;
import android.view.View;

import androidx.annotation.NonNull;

public class HoverPushOpacityEffect extends HoverPushEffect {
    private final float opacity = 0.6f;

    public HoverPushOpacityEffect() {
        this(MEDIUM);
    }

    public HoverPushOpacityEffect(@NonNull Pressure pressure) {
        super(pressure);
    }

    @Override
    protected void down(@NonNull View v) {
        super.down(v);
        animate(v, 1f, opacity);
    }

    @Override
    protected void up(@NonNull View v) {
        super.up(v);
        animate(v, opacity, 1f);
    }

    private void animate(View v, float from, float to) {
        ValueAnimator animator = ValueAnimator.ofFloat(from, to);
        animator.setDuration(100);
        animator.addUpdateListener(animation -> {
            float value = (float) animation.getAnimatedValue();
            v.setAlpha(value);
        });
        animator.start();
    }
}

File: app/src/main/java/com/quranapp/android/utils/gesture/HoverPushOpacityEffect.kt
package com.quranapp.android.utils.gesture

import android.animation.ValueAnimator
import android.view.View

class HoverPushOpacityEffect(pressure: Pressure = Pressure.MEDIUM) : HoverPushEffect(pressure) {
    private val opacity = 0.6f

    override fun down(v: View) {
        super.down(v)
        animate(v, 1f, opacity)
    }

    override fun up(v: View) {
        super.up(v)
        animate(v, opacity, 1f)
    }

    private fun animate(v: View, from: Float, to: Float) {
        ValueAnimator.ofFloat(from, to).apply {
            duration = 100
            addUpdateListener { animation: ValueAnimator ->
                val value = animation.animatedValue as Float
                v.alpha = value
            }
        }.start()
    }
}
